Server/src/models/AuthenticationModel.ts

```diff
@@ -0,0 +1,108 @@
+import { getConnection } from 'typeorm';
+
+import { Accounts } from './entities/Accounts'
+import { ISignUpInformation } from '../genericInterfaces/AuthenticationInterfaces'
+
+/**
+ * This model contains all methods required for obtaining data from the Accounts
+ * table inside the database
+ */
+export class AuthenticationModel {
+
+    /**
+     * Method responsible for completing signup process. Extracting user information
+     * and sets the entity values
+     * @param signUpInfo - User Information from Frontend Request
+     */
+    static async insertSignUpInformation(signUpInfo: ISignUpInformation) {
+
+        let connection = getConnection();
+
+        let signUpInformation = new Accounts();
+        signUpInformation.id;
+        signUpInformation.email = signUpInfo.email;
+        signUpInformation.password = signUpInfo.password;
+        signUpInformation.firstName = signUpInfo.firstName;
+        signUpInformation.lastName = signUpInfo.lastName;
+        signUpInformation.dateOfBirth = signUpInfo.dateOfBirth;
+        signUpInformation.organizationName = signUpInfo.organizationName;
+        signUpInformation.admin = signUpInfo.isAdmin;
+        await connection.manager.save(signUpInformation);
+    }
+
+    /**
+     * Method to verify Email exists. Returns true or undefined
+     * @param email - User Email
+     */
+    static async verifyIfEmailExists(email: string): Promise<boolean> {
+
+        let connection = getConnection();
+        let userEmail = await connection.getRepository(Accounts)
+            .createQueryBuilder('accounts')
+            .where('accounts.email = :email', { email: email })
+            .getOne()
+
+        return userEmail !== undefined;
+    }
+
+    /**
+     * Method used to verify Password of corresponding input email
+     * @param email - User Email
+     */
+    static async verifyPassword(email: string): Promise<any> {
+
+        let connection = getConnection();
+        let userInfo = await connection.manager
+            .createQueryBuilder(Accounts, 'account')
+            .select('account.email', 'account_email')
+            .select('account.password', 'account_password')
+            .where('account.email = :email', { email: email })
+            .getRawMany();
+        return userInfo[0].account_password;
```

Comment:
not a password again, a hash

Reply:
i would call this function "getPasswordHash" cause no actual verification is done